communication/src/main/java/org/jboss/da/communcation/pnc/authentication/PNCAuthentication.java

```diff
@@ -0,0 +1,133 @@
+        return connect(url, urlParams);
+    }
+
+    public static String getAccessToken(String url, String clientId, String username, String password)
+            throws ClientProtocolException, IOException {
+        Map<String, String> urlParams = new HashMap<String, String>();
+        urlParams.put("grant_type", "password");
+        urlParams.put("client_id", clientId);
+        urlParams.put("username", username);
+        urlParams.put("password", password);
+        return connect(url, urlParams)[0];
+    }
+
+    public static String getrefreshToken(String url, String clientId, String username, String password)
+            throws ClientProtocolException, IOException {
+        Map<String, String> urlParams = new HashMap<String, String>();
+        urlParams.put("grant_type", "password");
+        urlParams.put("client_id", clientId);
+        urlParams.put("username", username);
+        urlParams.put("password", password);
+        return connect(url, urlParams)[1];
+    }
+
+    private static String[] connect(String url, Map<String, String> urlParams)
+            throws ClientProtocolException, IOException {
+
+        CloseableHttpClient httpclient = HttpClients.createDefault();
+        HttpPost httpPost = new HttpPost(url);
+
+        // add header
+        httpPost.setHeader("Content-Type", "application/x-www-form-urlencoded");
+
+        List<BasicNameValuePair> urlParameters = new ArrayList<BasicNameValuePair>();
+        for (String key : urlParams.keySet()) {
+            urlParameters.add(new BasicNameValuePair(key, urlParams.get(key)));
+        }
+        httpPost.setEntity(new UrlEncodedFormEntity(urlParameters));
+        CloseableHttpResponse response = httpclient.execute(httpPost);
+
+        String refreshToken = "";
+        String accessToken = "";
+        try {
+            BufferedReader rd = new BufferedReader(
+                    new InputStreamReader(response.getEntity().getContent()));
+
+            String line = "";
+            while ((line = rd.readLine()) != null) {
+                if (line.contains("refresh_token")) {
+                    String[] respContent = line.split(",");
+                    for (int i = 0; i < respContent.length; i++) {
+                        String split = respContent[i];
+                        if (split.contains("refresh_token")) {
+                            refreshToken = split.split(":")[1].substring(1, split.split(":")[1].length() - 1);
+                        }
+                        if (split.contains("access_token")) {
+                            accessToken = split.split(":")[1].substring(1, split.split(":")[1].length() - 1);
+                        }
+                    }
+                }
+            }
+        } finally {
+            response.close();
+        }
+        return new String[]{accessToken, refreshToken};
+
+
+    }
+
+    public String authenticate() {
+        try(DAConfig conf = config.getConfig()) {
```

Comment:
There is no point for using try-with-resources, as DAConfig shouldn't be closable.